lib/peer_manager/modules/nrf_ble_lesc.c

```diff
@@ -118,35 +106,55 @@ uint32_t nrf_ble_lesc_keypair_generate(void)
 	m_keypair_generated = false;
 	m_lesc_oobd_own_generated = false;
 
-	LOG_DBG("Generating ECC key pair");
-	err_code = nrf_crypto_ecc_key_pair_generate(&m_keygen_context,
-						    &g_nrf_crypto_ecc_secp256r1_curve_info,
-						    &m_private_key, &m_public_key);
-	if (err_code != NRF_SUCCESS) {
-		LOG_ERR("nrf_crypto_ecc_key_pair_generate() returned error 0x%x.", err_code);
-		return err_code;
+	/* Destroy the previous key pair (if any), to free up memory. */
+	status = psa_destroy_key(m_keypair_id);
+	if (status != PSA_SUCCESS && status != PSA_ERROR_INVALID_HANDLE) {
+		LOG_ERR("psa_destroy_key returned status %d", status);
+	} else {
+		m_keypair_id = 0;
 	}
 
-	/* Convert to a raw type. */
-	err_code =
-		nrf_crypto_ecc_public_key_to_raw(&m_public_key, m_lesc_public_key.pk, &public_len);
-	if (err_code != NRF_SUCCESS) {
-		LOG_ERR("nrf_crypto_ecc_public_key_to_raw() returned error 0x%x.", err_code);
-		return err_code;
+	LOG_DBG("Generating ECC key pair");
+
+	psa_key_attributes_t key_attributes = PSA_KEY_ATTRIBUTES_INIT;
+
+	psa_set_key_usage_flags(&key_attributes, PSA_KEY_USAGE_DERIVE);
+	psa_set_key_lifetime(&key_attributes, PSA_KEY_LIFETIME_VOLATILE);
+	psa_set_key_algorithm(&key_attributes, PSA_ALG_ECDH);
+	psa_set_key_type(&key_attributes, PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1));
+	psa_set_key_bits(&key_attributes, 256);
+
+	status = psa_generate_key(&key_attributes, &m_keypair_id);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_generate_key() returned status %d", status);
+		return NRF_ERROR_INTERNAL;
 	}
 
-	/* Invert the raw type to little-endian (required for BLE). */
-	err_code = nrf_crypto_ecc_byte_order_invert(&g_nrf_crypto_ecc_secp256r1_curve_info,
-						    m_lesc_public_key.pk, m_lesc_public_key.pk,
-						    NRF_CRYPTO_ECC_SECP256R1_RAW_PUBLIC_KEY_SIZE);
-	if (err_code != NRF_SUCCESS) {
-		LOG_ERR("nrf_crypto_ecc_byte_order_invert() returned error 0x%x.", err_code);
+	/* Export the raw representation of the public key. */
+	status = psa_export_public_key(m_keypair_id, pub_key, sizeof(pub_key), &pub_key_len);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_export_public_key() returned status %d", status);
+		return NRF_ERROR_INTERNAL;
+
+	} else if (pub_key_len != sizeof(pub_key)) {
+		LOG_ERR("psa_export_public_key() unexpected output length %d", pub_key_len);
```

Comment:
I don't think that you need a run time check for this to be honest.  The PSA apis specify both the size of the  output and the representation. So this seems redundant and the check below about the pub_key first byte being 0x4 as well.

Reply:
Thanks. Removed the extra checks.